ethers/providers/jsonrpc/subscriptions.nim

```diff
@@ -14,6 +14,14 @@ type
     callbacks: Table[JsonNode, SubscriptionCallback]
   SubscriptionCallback = proc(id, arguments: JsonNode) {.gcsafe, raises:[].}
 
+# FIXME Nim 1.6.XX seems to have issues tracking exception effects and will see
```

Comment:
Actually, this is probably the wrong way of doing it, as you noted in the forum post, annotating the callback type with a `raises: []` fixes the probable, this is in fact the correct fix, callbacks should (almost) never allow exceptions to escape.

Reply:
Also, here is a pointer on how to handle the strict effect changes in a backwards/forwards compatible way in the 1.6.0 release notes - https://github.com/narimiran/Nim/blob/f14925d3bc778bcb7eef4eb9085f08d65f4281ce/changelogs/changelog_1_6_0.md#strict-effects, should be helpful.

Reply:
In summary, the solution would be a combination of using `raises:[]` in the callback definition, and changing the callback table key type.

Reply:
Note that table accessors such as `get` or `[]` will raise an exception, so this should either use the non throwing versions (getOrDefault, etc), or handle the exceptions in the callback itself.

Reply:
I love when two different issues conflate into the same symptom. :-D I'm not sure I follow your explanation though @emizzle (and that's probably the problem :-)): is parseJson called during the hashing of JsonNode?

@dryajov yes but in this case we're writing to the table (`[]=`), this should not raise exceptions.